diff --git a/kernel/Makefile b/kernel/Makefile
index aa380e9a..48ffbed3 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -65,9 +65,15 @@ endif
 $(info -- KernelSU-Next Manager signature size: $(KSU_NEXT_EXPECTED_SIZE))
 $(info -- KernelSU-Next Manager signature hash: $(KSU_NEXT_EXPECTED_HASH))
 
+$(info -- KernelSU Manager signature size: $(KSU_EXPECTED_SIZE))
+$(info -- KernelSU Manager signature hash: $(KSU_EXPECTED_HASH))
+
 ccflags-y += -DEXPECTED_NEXT_SIZE=$(KSU_NEXT_EXPECTED_SIZE)
 ccflags-y += -DEXPECTED_NEXT_HASH=\"$(KSU_NEXT_EXPECTED_HASH)\"
 
+ccflags-y += -DEXPECTED_SIZE=$(KSU_EXPECTED_SIZE)
+ccflags-y += -DEXPECTED_HASH=\"$(KSU_EXPECTED_HASH)\"
+
 ccflags-y += -DKSU_COMPAT_GET_CRED_RCU
 
 ccflags-y += -DKSU_UMOUNT
@@ -150,4 +156,4 @@ endif
 ccflags-y += -Wno-implicit-function-declaration -Wno-strict-prototypes -Wno-int-conversion -Wno-gcc-compat
 ccflags-y += -Wno-declaration-after-statement -Wno-unused-function
 
-# Keep a new line here!! Because someone may append config
+# Keep a new line here!! Because someone may append config
\ No newline at end of file
diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 384bb1c5..5fbae337 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -316,5 +316,11 @@ module_param_cb(ksu_debug_manager_uid, &expected_size_ops,
 
 bool is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH);
+	return (check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH) 
+	// Rissu KernelSU Fork
+	|| check_v2_signature(path, 0x396, f415f4ed9435427e1fdf7f1fccd4dbc07b3d6b8751e4dbcec6f19671f427870b) 
+	// MKSU 5ec1cff 💜
+    || check_v2_signature(path, 384, 7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4) 
+	// KernelSU Official
+	|| check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH));
 }
\ No newline at end of file
